phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -478,31 +490,65 @@ public ConnectionQueryServicesImpl(QueryServices services, ConnectionInfo connec
 
         // Without making a copy of the configuration we cons up, we lose some of our properties
         // on the server side during testing.
-        this.config = HBaseFactoryProvider.getConfigurationFactory().getConfiguration(config);
+        Configuration finalConfig = HBaseFactoryProvider
```

Comment:
Can we add some comments to the effect, that this allows the application overridden ConfigurationFactory to inject/modify configs